### PR TITLE
8277372: Add getters for BOT and card table members

### PR DESCRIPTION
**Pull request by vish-chan**
Changed the visibility, added getters and refactored the following:

1. Card Table Members
2. BOT members
3. ObjectStartArray block members

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277372](https://bugs.openjdk.java.net/browse/JDK-8277372): Add getters for BOT and card table members

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 0f63516244b3db50b7bfd392b274ae034f070902
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to 0f63516244b3db50b7bfd392b274ae034f070902

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6570/head:pull/6570` \
`$ git checkout pull/6570`

Update a local copy of the PR: \
`$ git checkout pull/6570` \
`$ git pull https://git.openjdk.java.net/jdk pull/6570/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6570`

View PR using the GUI difftool: \
`$ git pr show -t 6570`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6570.diff">https://git.openjdk.java.net/jdk/pull/6570.diff</a>

</details>

### CODE REVIEW

**Comment by tschatzl** [comment]
@tstuefe : can you check whether the s390 and ppc changes still compile? The changes look straightforward enough, but...

Thanks,
  Thomas

**Comment by tstuefe** [comment]
@tschatzl We have s390 + ppcle builds now in GHAs thanks to Alexey, and they do look fine (https://github.com/openjdk/jdk/pull/6570/checks?check_run_id=4341321057). Seeing how simple the platform changes are, I think this is okay. 

Cheers, Thomas

File: src/hotspot/share/gc/shared/blockOffsetTable.hpp [hunk 1]
```diff
@@ -49,12 +49,13 @@
...
-public:
-  static uint LogN;
-  static uint LogN_words;
-  static uint N_bytes;
-  static uint N_words;
+private:
+  static uint _LogN;
+  static uint _LogN_words;
+  static uint _N_bytes;
+  static uint _N_words;
```
Comment (tschatzl) [inline src/hotspot/share/gc/shared/blockOffsetTable.hpp:56]: The `private` visibility modifier can be removed as this is default at the top of a class.
The static variables should start with a lower case letter after the underscore, something like `_log_n`.

My suggestion would also be to change `N`/`n` to something more understandable, like `size`, and add `block`, i.e. something like `_log_block_size`, `_log_block_size_in_words` similar to the corresponding `CardTable` members etc.

**Review by tschatzl — CHANGES_REQUESTED** [review]
I will push it through our testing, particular the changes for the SA agent (in `vmstructs_gc.hpp`) are always good to double-check.

File: src/hotspot/share/gc/shared/blockOffsetTable.hpp (hunk 1 above)
Comment (vish-chan) [reply src/hotspot/share/gc/shared/blockOffsetTable.hpp:56]: As I can understand, we need to replace "N" with something meaningful. Does something like "entry_size" or "bot_entry_size" would work?

File: src/hotspot/share/gc/parallel/objectStartArray.hpp [hunk 2]
```diff
@@ -46,16 +46,16 @@ class ObjectStartArray : public CHeapObj<mtGC> {
   jbyte*          _raw_base;
   jbyte*          _offset_base;
 
+  static uint _block_shift;
+  static uint _block_size;
+  static uint _block_size_in_words;
+
```
[the remark this answers is not in the document]
Comment (vish-chan) [reply src/hotspot/share/gc/parallel/objectStartArray.hpp:52]: Shall these be renamed to osa_card_shift? Something like this?

[the remark this answers is not in the document]
Comment (tschatzl) [reply src/hotspot/share/gc/parallel/objectStartArray.hpp:52]: Okay, '_osa_card_shift` and so on is also good.

[the remark this answers is not in the document]
Comment (vish-chan) [reply src/hotspot/share/gc/parallel/objectStartArray.hpp:52]: You mean for both blockOffsetTable and objectStartArray, all the relevant members be renamed to _card_size and so on?

[the remark this answers is not in the document]
Comment (kstefanj) [reply src/hotspot/share/gc/parallel/objectStartArray.hpp:52]: I think this sounds like a good plan.

**Review by kstefanj — CHANGES_REQUESTED** [review]
Nice to see this unified. Some comments below.

**Review by albertnetymk — APPROVED** [review]
Just a very minor comment.

**Review by kstefanj — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Comment by vish-chan** [comment]
@tschatzl is there anything else required from my end?

**Comment by tschatzl** [comment]
/sponsor
Thanks @vish-chan for fixing all this.

**Pull request closed without merging**